src/network/networkd-link.c

```diff
@@ -261,6 +261,20 @@ static int link_enable_ipv6(Link *link) {
         return 0;
 }
 
+static bool link_is_bridge_port(Link *link) {
+
+        assert(link);
+
+        if (link->flags & IFF_LOOPBACK)
+                return false;
+
+        if (link->network)
+                if (link->network->bridge)
+                        return true;
```

Comment:
I don't get your point. I didn't mean to replace `return link->bridge_port;` with what I quoted. I mean:

```
static bool link_is_bridge_port(Link *link) {

        assert(link);

        if (link->flags & IFF_LOOPBACK)
                return false;

        if (!link->network)
                return false;

        if (link->network->bridge)
                return false;

        return link->bridge_port;
}
```

It pretty much does exactly what your code did, except it return false it the link is a bridge, or return link->bridge_port if it is not a bridge. (I don't see how you would want to return true if the link is a bridge.)